media/windows/ECC-Ecobee/ECC Pycobee Data.py

```diff
@@ -85,12 +93,37 @@
 from pythonping import ping
 
 # Define version
-eccpycobee_version = "02.09"
-eccpycobee_date = "02-Jan-2020"
+eccpycobee_version = "03.01"
+eccpycobee_date = "06-Jan-2020"
+
+# Parse the command line arguments for the filename locations, if present
+parser = argparse.ArgumentParser(description='''Epiphany Catholic Church Ecobee Thermostat Polling Application.
+                                            This routine will poll information from the Ecobee thermostats and
+                                            write the data to a SQLite3 database.''',
+                                 epilog='''Filename parameters may be specified on the command line at invocation, 
+                                        or default values will be used for each.''')
+parser.add_argument("-l", "-log", "--log_file_path", default="ECCEcobee.log",
```

Comment:
```suggestion
parser.add_argument("--log", default="ECCEcobee.log",
```

Reply:
I probably don't understand why you're recommending the change; is using the "--param" the most common method of specifying command-line arguments?  Otherwise, I see these as just options...as in pick the one you want to use (if any)??

Reply:
Yeah, sorry, not trying to be a jerk here -- I was just trying to make easy-to-apply suggestions.  Here's my rationale:

- it's common Linux convention to have `-` (single dash) followed by a series of single-character options.  E.g., `-a`, `-b`, etc.  And if there's multiple characters, it's assumed that it's the union of several single-character options.  I.e., `-ab` is the equivalent of `-a -b`.
- Additionally, `--` (double dash) can use "long" option names, such as words or multiple words.  E.g., `--param`.  It's also common to try to use `-` instead of `_` to separate words.  For example: `--log-file-path` instead of `--log_file_path`.  The argument here is that is just slightly easier to type all `-` rather than alternate between `-` and `_`.
- Finally, I tend prefer the "long" names (vs. single-letter options), mainly because I have a terrible memory.  Meaning: I like to have a command line like this:
```sh
$ ./myprogram --file foo.txt --url http://cnet.com --safe --locking
# vs.
$ ./myprogram -F foo.txt -u http://cnet.com -s -l
```
Just on the argument that when I look at that script again in 6 months, I'll have a better chance of remembering what those options are for without having to look them up again.

That being said, my suggestions were also infused with some personal preference -- e.g., `--log` vs. `--log-file-path`, etc.

Moral of the story here: feel free to take my suggestions or not.  😄 They are purely suggestions!

Reply:
Ah, I see more clearly now... Don't get me wrong; I appreciate the input.  But, I also like to understand the reasoning behind suggested changes.  To me there's a distinction between personal preferences and conventions / standards, and I'm still feeling my way around some of the common conventions...many of which are not always clearly documented somewhere.

I'll sort through these and merge in your suggestions.  Will probably leave the variations though.

Reply:
Roger!  👍